src/main/java/pl/cleankod/exchange/entrypoint/AccountController.java

```diff
@@ -25,35 +27,35 @@ public AccountController(FindAccountAndConvertCurrencyUseCase findAccountAndConv
     }
 
     @GetMapping(path = "/{id}")
-    public ResponseEntity<Account> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
+    ResponseEntity<AccountDto> findAccountById(@PathVariable String id, @RequestParam(required = false) String currency) {
         return Optional.ofNullable(currency)
                 .map(s ->
                         findAccountAndConvertCurrencyUseCase.execute(Account.Id.of(id), Currency.getInstance(s))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
+
                 )
                 .orElseGet(() ->
                         findAccountUseCase.execute(Account.Id.of(id))
-                                .map(ResponseEntity::ok)
-                                .orElse(ResponseEntity.notFound().build())
-                );
+
+                )
+                .map(s -> new AccountDto(s.id().value(), s.number().value(), new MoneyDto(s.balance().amount(), s.balance().currency())))
```

Comment:
I forgot to mark this PR as a Draft (I have added a prefix to it) - most of the changes have not been pushed yet.
I will remove Draft prefix as soon as it will be ready (probably tomorrow)

Reply:
I changed the type to "Draft". When you're done, use the "Ready for review" button and let us know.

Reply:
Thank you. Pull review is ready to review.